docs/okeps/okep-5233-pre-assigned-pod-net-id.md

```diff
@@ -0,0 +1,378 @@
+// It is not recommended to set this field without explicit need and understanding of the OVN network topology.
+// When omitted, the first IP address from the pod network subnet is used.
+//
+// +optional
+DefaultGatewayIPs DualStackIPs `json:"defaultGatewayIPs,omitempty"`
+
+// NodeManagementIPs are is the node management IP used in the internal OVN topology.
+//
+// Dual-stack clusters may set 2 IPs (one for each IP family), otherwise only 1 IP is allowed.
+// This field is only allowed for "Primary" network.
+// It is not recommended to set this field without explicit need and understanding of the OVN network topology.
+// When omitted, the second IP address from the pod network subnet is used.
+//
+// +optional
+NodeManagementIPs DualStackIPs `json:"nodeManagementIPs,omitempty"`
+
+// IPAM section contains IPAM-related configuration for the network.
+// +optional
+IPAM *IPAMConfig `json:"ipam,omitempty"`
+}
+
+// +kubebuilder:validation:XValidation:rule="isIP(self)", message="IP is invalid"
+type IP string
+
+// +kubebuilder:validation:MinItems=1
+// +kubebuilder:validation:MaxItems=2
+// +kubebuilder:validation:XValidation:rule="size(self) != 2 || !isIP(self[0]) || !isIP(self[1]) || ip(self[0]).family() != ip(self[1]).family()", message="When 2 IPs are set, they must be from different IP families"
+type DualStackIPs []IP
+
+```
+
+#### IPAMClaim API changes
+
+- TODO: Add status conditions to the IPAMClaim CRD, this will be an external enhancement in the network-plumbing group
+
+### Implementation Details
+
+#### Configurability
+
+The changes outlined in this enhancement should be configurable. This means a configuration knob
+is required to instruct OVN-Kubernetes on whether to process the annotation described in the 
+[Pod network identity](#pod-network-identity) section.
+
+#### NetworkSelectionElement annotation
+
+The `v1.multus-cni.io/default-network` annotation should only be processed for new pods, 
+modifying it after the addresses were allocated won't be reflected in the pods network 
+configuration and this should be blocked through a
+[Validating Admission Policy](https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/):
+```
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: pre-assigned-network-identity
+spec:
+  matchConstraints:
+    resourceRules:
+      - apiGroups:   [""]
+        apiVersions: ["v1"]
+        operations:  ["UPDATE"]
+        resources:   ["pods"]
+  failurePolicy: Fail
+  validations:
+    - expression: "('v1.multus-cni.io/default-network' in oldObject.metadata.annotations) == ('v1.multus-cni.io/default-network' in object.metadata.annotations)"
+      message: "The 'v1.multus-cni.io/default-network' annotation cannot be changed after the pod was created"
+```
+The `NetworkSelectionElement` structure has an extensive list of fields, this enhancement 
+focuses only on the following:
+```cgo
+type NetworkSelectionElement struct {
```

Comment:
can you show a full example of a user creating a pod with the network selection element filled out and the multus annotation? Maybe in your workflow/proposed solution section.